nautobot_device_lifecycle_mgmt/software.py

```diff
@@ -0,0 +1,149 @@
+"""Django classes and functions handling Software Lifecycle related functionality."""
+
+from datetime import date
+
+from django.db.models import Case, IntegerField, Q, Value, When
+from django.contrib.contenttypes.models import ContentType
+from nautobot.extras.models import RelationshipAssociation
+
+from .models import ValidatedSoftwareLCM, SoftwareLCM
+from .tables import ValidatedSoftwareLCMTable
+
+
+class ItemSoftware:
```

Comment:
Would it make sense to convert `get_validated_software_qs` into a queryset class for `ValidatedSoftwareLCM`
This would allow us to have something like that 
```python
device = Device.objects.get(XXX)
validated_software = ValidatedSoftwareLCM.objects.get_for_model(device)
```
it's easier than it look to implement
We just need to create a new class 
```python
class ValidatedSoftwareLCMQuerySet(RestrictedQuerySet):
    """Queryset for `ValidatedSoftwareLCM` objects."""

    def get_for_object(self, obj):
        """
        Return all `ValidatedSoftwareLCM` assigned to the given object.
        """
       if obj is of type Device:
             do this
       else:
             do that

       [..]

class ValidatedSoftwareLCM([..]):
      objects = ValidatedSoftwareLCMQuerySet.as_manager()
```


Here is an example https://github.com/nautobot/nautobot/pull/1047/files#diff-01ee6b16cefc4c5378473d290760c32b2507d31249d3df90972f4dca6a93d33aR16

Reply:
I made an attempt at implementing that and moved a few things around. Seems to work well.

@dgarros @josh5276 @jvanderaa Could you please review and we could hopefully merge it soon.